src/Gui/Clipping.cpp

```diff
@@ -260,147 +394,148 @@ void Clipping::reject()
 }
 
 void Clipping::onFlipClipYClicked()
 {
-    d->flipY = !d->flipY;
-    SbPlane pln = d->clipY->plane.getValue();
-    d->clipY->plane.setValue(SbPlane(-pln.getNormal(), -pln.getDistanceFromOrigin()));
+    Private::activeClip->flipY = !Private::activeClip->flipY;
+    SbPlane pln = Private::activeClip->clipY->plane.getValue();
+    Private::activeClip->clipY->plane.setValue(SbPlane(-pln.getNormal(), -pln.getDistanceFromOrigin()));
 }
 
 void Clipping::onFlipClipZClicked()
 {
-    d->flipZ = !d->flipZ;
-    SbPlane pln = d->clipZ->plane.getValue();
-    d->clipZ->plane.setValue(SbPlane(-pln.getNormal(), -pln.getDistanceFromOrigin()));
+    Private::activeClip->flipZ = !Private::activeClip->flipZ;
+    SbPlane pln = Private::activeClip->clipZ->plane.getValue();
+    Private::activeClip->clipZ->plane.setValue(SbPlane(-pln.getNormal(), -pln.getDistanceFromOrigin()));
 }
 
 void Clipping::onGroupBoxViewToggled(bool on)
 {
     if (on) {
-        d->ui.groupBoxX->setChecked(false);
-        d->ui.groupBoxY->setChecked(false);
-        d->ui.groupBoxZ->setChecked(false);
+        Private::ui.groupBoxX->setChecked(false);
+        Private::ui.groupBoxY->setChecked(false);
+        Private::ui.groupBoxZ->setChecked(false);
     }
 
-    d->clipView->on.setValue(on);
+    Private::activeClip->clipView->on.setValue(on);
 }
 
 void Clipping::onClipViewValueChanged(double val)
 {
-    SbPlane pln = d->clipView->plane.getValue();
-    d->clipView->plane.setValue(SbPlane(pln.getNormal(), val));
+    SbPlane pln = Private::activeClip->clipView->plane.getValue();
+    Private::activeClip->clipView->plane.setValue(SbPlane(pln.getNormal(), val));
 }
 
 void Clipping::onFromViewClicked()
 {
-    if (d->view) {
-        Gui::View3DInventorViewer* view = d->view->getViewer();
-        SbVec3f dir = view->getViewDirection();
-        SbPlane pln = d->clipView->plane.getValue();
-        d->clipView->plane.setValue(SbPlane(dir, pln.getDistanceFromOrigin()));
+    if (Private::activeClip) {
+        Gui::View3DInventorViewer* viewer = Private::activeClip->view->getViewer();
+        SbVec3f dir = viewer->getViewDirection();
+        SbPlane pln = Private::activeClip->clipView->plane.getValue();
+        Private::activeClip->clipView->plane.setValue(SbPlane(dir, pln.getDistanceFromOrigin()));
     }
 }
 
 void Clipping::onAdjustViewdirectionToggled(bool on)
 {
-    d->ui.dirX->setDisabled(on);
-    d->ui.dirY->setDisabled(on);
-    d->ui.dirZ->setDisabled(on);
-    d->ui.fromView->setDisabled(on);
+    Private::ui.dirX->setDisabled(on);
+    Private::ui.dirY->setDisabled(on);
+    Private::ui.dirZ->setDisabled(on);
+    Private::ui.fromView->setDisabled(on);
 
+    Private::activeClip->isSensing = on;
     if (on) {
-        d->sensor->schedule();
+        Private::activeClip->sensor->schedule();
     }
     else {
-        d->sensor->unschedule();
+        Private::activeClip->sensor->unschedule();
```

Comment:
If the user never closes the clipping plane dialog, but closes a view with the sensor scheduled ("custom dir; adjust to view dir"), the sensor is never unscheduled until the clipping plane dialog is closed.

Perhaps the `if` in `moveCallback` should have an `else` that unschedules the sensor, and even clean up the view's `Clipping::Private` object?  Maybe there's a way for the `Command` to be evented when a view is closed and take care of it all that way.

Reply:
I went ahead and implemented this; seem correct to me.